### DISCUSSION TITLE
Remote jupyter server for cell rendering (like vscode already supports)

### DISCUSSION THREAD

**Discussion by fradav** [quarto-dev/quarto-cli#2829] [Feature Requests]
Hi all,

Visual Studio Code already supports running “in-editor” cells on remote jupyter just by specifying an url to a jupyter server directly

![image](https://user-images.githubusercontent.com/701587/195567092-68205426-5247-4085-91b9-f919be513104.png)

As for quarto, my understanding is it isn’t possible. There are use cases where it would be very practical or even necessary to edit/render everything but jupyter cells by a locally installed quarto, and also render jupyter cells in a remote server, who sometimes doesn’t even offer the possibility to run or install quarto (Jupyterhub containerized instances for example).

I am a member of the staff of an online journal centered on reproductible research and we use quarto as the mandatory format and tooling for submissions. We would like to offer our authors and editors the possibility to run computations inside article submissions on external (possibly heavily equipped, cluster-like or with gpu) computational ressources, and let authors/editors handle the editing/rendering on their local computers.

Regards,

**Comment by fradav** [comment]
Possible tasks:
- [ ] Give a new configuration option for jupyter like `external-server`
- [ ] Modify `src/resources/jupyter/jupyter/jupyter.py` to take this option into account (function `execute_server` and inner class)
- [ ] Modify the whole setup logic in `src/resources/jupyter/lang` as it has been designed for local kernel execution (with local current directory and so on)

The third task should probably a non-trivial one.

**Comment by mcanouil** [comment]
Closing as completed and in favour of https://github.com/quarto-dev/quarto-cli/issues/2852
